File: 11340.py
import sys

def main():
    dic = list(range(257))
    casos = int(input())
    while casos > 0:
        qtLetras = int(input())
        for k in dic:
            k = 0
        valor = 0
        while qtLetras > 0:
            lt = input().split(' ')
            dic[ord(lt[0])] = int(lt[1])
            qtLetras -=1
        
        texto = int(input())
        while texto > 0:
            linha = input()
            for i in linha:
                if i != ' ':
                    valor += dic[ord(i)]
            texto -=1
        casos -=1
        valor = valor/100
        print("%.2f$"%valor)
        

if __name__ == "__main__":
    main()

File: 11340/11340.cpp
#include <cstdio>
#include <iostream>
#include <cstring>

using namespace std;


int main(){
    int casos,qtLetras,peso,texto;
    unsigned char let,letra;
    string linha;
    int ascii[256];
    unsigned long long valor;

    

    scanf("%d\n",&casos);
    while (casos > 0) {
        valor = 0;
        
        scanf("%d\n",&qtLetras);
        for(int i=0; i<256; i++){
            ascii[i] = 0;
        }
        
        while (qtLetras > 0){
            scanf("%c %d\n",&let,&peso);
            int v = (int)let;
            ascii[v] = peso;
            
            qtLetras --;
        }



        scanf("%d\n",&texto);
        while (texto > 0){
        
            scanf("%c",&letra);
			while ( letra != '\n') {
				valor += ascii[(int) letra];
                scanf("%c",&letra);
			}

            texto--;
        }
        casos --;
        

        printf("%.2f$\n",valor / 100.0);
        
    }


    return 0;
}